console/console-init/ui/src/modules/device/components/DeviceFilterCriteria/DeviceFilterCriteria.tsx

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import { IDropdownOption, DropdownWithToggle } from "components";
+import {
+  InputGroup,
+  TextInput,
+  Button,
+  Grid,
+  GridItem
+} from "@patternfly/react-core";
+import { MinusCircleIcon } from "@patternfly/react-icons";
+
+export enum operator {
+  lte = "lessthanequalto",
+  gte = "greaterthanequalto",
+  lt = "lessthan",
+  gt = "greaterthan",
+  eq = "equalto",
+  neq = "notequalto"
+}
+
+export const operatordropdownOptions: IDropdownOption[] = [
+  { key: operator.eq, value: operator.eq, label: "=" },
+  { key: operator.neq, value: operator.neq, label: "≠" },
+  { key: operator.gte, value: operator.gte, label: "≥" },
+  { key: operator.gt, value: operator.gt, label: ">" },
+  { key: operator.lte, value: operator.lte, label: "≤" },
+  { key: operator.lt, value: operator.lt, label: "<" }
+];
+
+export interface IDeviceFilterCriteria {
+  parameter: string;
+  operator: string;
+  value: string;
+  key: string;
+}
+interface IDeviceFilterCriteriaProps {
+  criteria: IDeviceFilterCriteria;
+  deleteCriteria: (criteria: IDeviceFilterCriteria) => void;
+  setCriteria: (criteria: IDeviceFilterCriteria) => void;
+}
+const DeviceFilterCriteria: React.FunctionComponent<IDeviceFilterCriteriaProps> = ({
+  criteria,
+  deleteCriteria,
+  setCriteria
+}) => {
+  const onChangeParameter = (value: string) => {
+    const criteriaObj = criteria;
```

Comment:
always create a new copy of object in case need to modify. Array and object is reference type so it can be create issue.
either you can spread operator or Object.assign() for new copy.
const criteriaObj = {...criteria};
OR 
const criteriaObj = Object.assign(criteria);